packages/flutter_image/lib/network.dart

```diff
@@ -53,6 +54,21 @@ class NetworkImageWithRetry extends ImageProvider<NetworkImageWithRetry> {
   /// fetch attempts failed.
   final FetchStrategy fetchStrategy;
 
+  /// HTTP Headers to add to the request.
+  ///
+  /// These headers will override any existing headers used by the network
+  /// library. By default all headers are changed to lowercase before sending.
+  /// To preserve the case of the headers set [preserveHeaderCase] to true
+  final Map<String, Object>? requestHeaders;
+
+  /// Controls whether the [requestHeaders] passed along to the server are
+  /// kept as is or changed to lower case before sending.
+  ///
+  /// If true the case of the headers is kept exactly as is given in
+  /// [requestHeaders]. The default value is false, the networking library
+  /// changes all letters to lowercase.
+  final bool preserveHeaderCase;
```

Comment:
(Moot if the above is changed, but in case it's not:)

Why does this parameter exist? Anyone relying on specific behavior here is violating the HTTP spec. Do we have evidence that clients of this library need that behavior, such that adding this specific piece of API complexity is warranted?

Reply:
Sometimes script in languages like PHP might use case sensitive http headers ( which is not to spec of course). But for the sake of brevity, I've removed the parameter.